src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,269 @@
+        printLine();
+
+        for (int i = 0; i < tasks.size(); i++) {
+            System.out.print(i + 1);
+            System.out.println("." + tasks.get(i));
+        }
+
+        printLine();
+    }
+
+    public static void addTask(Task task) {
+        printLine();
+
+        tasks.add(task);
+        System.out.println("Got it. I've added this task:");
+        System.out.println(task);
+        System.out.println("Now you have " + tasks.size() + " task(s) in the list.");
+
+        printLine();
+    }
+
+    public static void addDeadline(String detail) {
+        int byIndex = detail.indexOf("/by");
+        boolean isByPresent = byIndex != -1;
+        if (isByPresent) {
+            String by = detail.substring(byIndex + 3).trim();
+            String deadlineDesc = detail.substring(0, byIndex).trim();
+            addTask(new Deadline(deadlineDesc, by));
+        } else {
+            System.out.println("Sorry, missing /by argument...");
+        }
+    }
+
+    public static void addEvent(String detail) {
+        int atIndex = detail.indexOf("/at");
+        boolean isAtPresent = atIndex != -1;
+        if (isAtPresent) {
+            String at = detail.substring(atIndex + 3).trim();
+            String eventDesc = detail.substring(0, atIndex).trim();
+            addTask(new Event(eventDesc, at));
+        } else {
+            System.out.println("Sorry, missing /at argument...");
+        }
+    }
+
+
+    public static boolean markTask(String response) {
+        printLine();
+
+        try {
+            String[] words = response.split(" ");
+            int taskIndex = Integer.parseInt(words[1]);
+            boolean isNotIndex = taskIndex > tasks.size() || taskIndex == 0 || taskIndex < 0;
+
+            if (isNotIndex) {
+                System.out.println("Sorry, seems like there's no such task with that index.");
+                printLine();
+                return false;
+            } else {
+                Task t = tasks.get(taskIndex - 1); //list indexing to the user starts from 1 but list indexing in fact starts from 0 internally
+                boolean isNotDone = !t.isDone;
+
+                if (isNotDone) {
+                    t.markAsDone();
+                    System.out.println("Nice! I've marked this task as done:");
+                    System.out.println(t);
+                } else {
+                    System.out.println("Hmm, you've already marked this task.");
+                }
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("☹ OOPS!!! You forgot to indicate an index!");
+        }
+
+        printLine();
+        return true;
+    }
+
+    public static boolean unmarkTask(String response) {
+        printLine();
+
+        try {
+            String[] words = response.split(" ");
+            int taskIndex = Integer.parseInt(words[1]);
+            boolean isNotIndex = taskIndex > tasks.size() || taskIndex == 0;
```

Comment:
Consider naming boolean variables in the positive form e.g isIndex to avoid potential confusion with double negatives.